README.md

```diff
@@ -134,30 +151,167 @@ or in the case of an error:
+  success: true;
+};
+```
+
+or in the case of an error:
+
+```ts
+type RegisterPublicKeyResponse = {
+  error: string;
+};
+```
+
+This route lets the user register public keys for their profile and/or set chain
+preferences for their registered public keys. The keys being registered must
+sign allowances (using the same nonce) to allow the profile to register them (or
+else anyone could claim any public key as their own). Thus, the keys being
+registered must set `allow` to the profile's public key that is registering
+them. `chainIds` is an optional array of chain IDs to set as preferences for
+this public key. If `chainIds` is omitted for a key being registered, it will
+just set the chain preference for the chain used to sign the nested request. The
+authentication structure is nested, so first the keys being registered need to
+sign the data objects, and then the profile public key performing the
+registration needs to sign all of them.
+
+All public keys attached to a profile will resolve to the profile when looked up
+using the GET request at the top (by address, public key, or bech32 hash).
+However, **a profile must explicitly choose a public key to use for name
+resolution/searching on each chain**. This is because chains sometimes use
+different public key derivation paths from other chains, and thus wallets
+aggregate default addresses that are associated with different public keys in
+the same wallet. Requiring an explicit opt-in per chain prevents the user from
+using an address for a profile that the profile holder does not expect to use.
+Although it is still safe to use any of the public keys controlled by the
+wallet/unique private key (since a user can manually add a new derivation path
+to their wallet), it would lead to unexpected behavior to use the undesired
+public key for a chain. Thus, the user needs to explicitly choose which public
+key (and thus which address) to resolve their name to on each chain.
+
+The nonce from the latest GET request must be provided to prevent replay
+attacks.
+
+The signature is derived by calling `OfflineAminoSigner`'s `signAmino` function
+with the `signDoc` argument generated using `makeSignDoc` from the
+`@cosmjs/amino` package. This can be seen in the signature verification code
+located in [src/index.ts](./src/utils/auth.ts#L20) around line 20.
+
+### `POST /unregister`
+
+The expected request body type is:
+
+```ts
+type UnregisterPublicKeyRequest = {
+  data: {
+    publicKeys: string[]
+    auth: {
+      type: string;
+      nonce: number;
+      chainId: string;
+      chainFeeDenom: string;
+      chainBech32Prefix: string;
+      publicKey: string;
+    };
+  };
+  signature: string;
+};
+```
+
+The returned type is:
+
+```ts
+type UnregisterPublicKeyResponse = {
+  success: true;
+};
+```
+
+or in the case of an error:
+
+```ts
+type UnregisterPublicKeyResponse = {
+  error: string;
+};
+```
+
+This route lets the user unregister public keys from their profile.
```

Comment:
Do we expect users to call this route? Is this something we'll call programmatically from a backend process? I guess I'm asking: when would this route be used?

Reply:
no idea, we're not going to make UI for it, but if you can register something, you should be able to unregister it too